libp2p/protocols/pubsub/pubsubpeer.nim

```diff
@@ -322,13 +322,11 @@ proc sendEncoded*(p: PubSubPeer, msg: seq[byte], isHighPriority: bool): Future[v
 
   if msg.len <= 0:
     debug "empty message, skipping", p, msg = shortLog(msg)
-    return
-
-  if msg.len > p.maxMessageSize:
+    Future[void].completed()
```

Comment:
why is this necessary?

Reply:
the function must return a Future on all return paths (it's no longer `async`, so there's no "implicit" creation of the future)